README.md

```diff
@@ -50,12 +50,23 @@ and you're welcome to give it a shot. Please review the installation page [here]
 
 ## Contributing to Vagrant
 
+Package dependencies: Vagrant requires `bsdtar` to be available on your system PATH to run tests successfully.
+
 Once your Vagrant bundle is installed from Git repository, you can run the test suite with:
 
     bundle exec rake
 
 This will run the unit test suite, which should come back all green!
 
+If you are developing Vagrant on a machine that already has a Vagrant package installation present, both will attempt to use the same folder for their configuration (location of this folder depends on system). In this case, override the
+`VAGRANT_HOME` environment variable for your development version of Vagrant before running any commands. For example, in Bash:
+
+    export VAGRANT_HOME=/path/to/project/.vagrant
+
+This folder will be ignored by Git. You can now run Vagrant commands against the development version:
+
+    bundle exec vagrant
+
```

Comment:
I'm okay with including this within the README as an optional step. This will generally only be useful when you have plugins installed (as the new version may require the plugins to be re-installed when being loaded). However, the `VAGRANT_HOME` directory should not be set to the project's local data directory (`.vagrant`). While it may work okay there is a good chance of unexpected behavior based on what data is being written to Vagrant's home directory and the project's local data directory. This would be better as a project local home (`.vagrant-home`) or something located in `/tmp`, etc.

Reply:
Yes, the plugin scenario is how I ran into it. I figured it would be a common enough scenario for new devs that a spot in the README was warranted. I'll update as requested.